x/ibctransfermiddleware/keeper/keeper.go

```diff
@@ -104,3 +109,111 @@ func (k Keeper) GetChannelFeeAddress(ctx sdk.Context, targetChannelID string) st
 	}
 	return channelFee.FeeAddress
 }
+
+type BridgeFee struct {
+	Fee      sdk.Coin
+	Sender   sdk.AccAddress
+	Receiver sdk.AccAddress
+}
+
+func (k Keeper) ChargeFee(ctx sdk.Context, msg *ibctypes.MsgTransfer) (*BridgeFee, error) {
+	params := k.GetParams(ctx)
+	// charge_coin := sdk.NewCoin(msg.Token.Denom, sdk.ZeroInt())
+	if params.ChannelFees != nil && len(params.ChannelFees) > 0 {
+		channelFee := findChannelParams(params.ChannelFees, msg.SourceChannel)
+		if channelFee != nil {
+			if channelFee.MinTimeoutTimestamp > 0 {
+
+				blockTime := ctx.BlockTime()
+
+				timeoutTimeInFuture := time.Unix(0, int64(msg.TimeoutTimestamp))
+				if timeoutTimeInFuture.Before(blockTime) {
+					return nil, fmt.Errorf("incorrect timeout timestamp found during ibc transfer. timeout timestamp is in the past")
+				}
+
+				difference := timeoutTimeInFuture.Sub(blockTime).Nanoseconds()
+				if difference < channelFee.MinTimeoutTimestamp {
+					return nil, fmt.Errorf("incorrect timeout timestamp found during ibc transfer. too soon")
+				}
+			}
+			coin := findCoinByDenom(channelFee.AllowedTokens, msg.Token.Denom)
+			if coin == nil {
+				return nil, fmt.Errorf("token not allowed to be transferred in this channel")
+			}
+
+			minFee := coin.MinFee.Amount
```

Comment:
Does this function takes all the fees or only some specific ones?

Reply:
function
calculate fee based on config as percentage fee + min fee for some specific demon

Reply:
Is this fee something different? Can't we use only it?

https://github.com/ComposableFi/composable-cosmos/blob/rustninja/pmf-middleware/custom/custompfm/keeper/keeper.go#L167

Reply:
No. this fee is the logic from original pfm.
it is different.

our requirements is to add extra field depends on channel and denom. where each fee contains 3 params
account where to send this fee, percentage from amount + min amount equal to 20$.